core/src/main/java/org/kiwix/kiwixmobile/core/search/viewmodel/State.kt

```diff
@@ -29,11 +29,26 @@ sealed class State {
   abstract val searchString: String
   abstract val searchOrigin: SearchOrigin
 
-  data class Results(
+  @Suppress("DataClassPrivateConstructor") // Make companion object act as constructor
+  data class Results private constructor(
     override val searchString: String,
     val values: List<SearchListItem>,
     override val searchOrigin: SearchOrigin
-  ) : State()
+  ) : State() {
+    companion object {
+      operator fun invoke(
+        searchString: String,
+        values: List<SearchListItem>,
+        searchOrigin: SearchOrigin
+      ): Results {
+        return Results(
+          searchString,
+          values.filter { it.value == searchString } + values.filter { it.value != searchString },
```

Comment:
I like this a bit less than what you had previously.

While this is quite terse it doesn't read well, I have to evaluate the code in my head to figure out what it does whereas
```
if (thisThing is in theList)
  thisThing + (theList - thisThing)
else
  theList
```
helps to describe the feature.

sidenote, this scenario is one where I think `filterNot` is applicable so the boolean expression can be reused and the reader doesn't have to reevaluate

Reply:
> thisThing

That doesn't work. `thisThing` can't be instantiated here because we don't know the concrete class type of `thisThing`. The `none {}` and the `filter {}` here avoid that problem.

I changed the last `filter` to `filterNot` as you requested.